Fix SDL2 sample build with latest vitasdk

bucanero opened the pull request:
after installing the latest vitasdk package available on GitHub, I tried to build the `redrectangle` SDL2 sample and I was getting the following linking error:

```
[ 16%] Linking C executable redrectangle
/usr/local/vitasdk/bin/../lib/gcc/arm-vita-eabi/10.3.0/../../../../arm-vita-eabi/bin/ld: /usr/local/vitasdk/arm-vita-eabi/lib/libSDL2.a(SDL_vitaaudio.c.obj): in function `VITAAUD_OpenCaptureDevice':
(.text+0x38): undefined reference to `sceAudioInOpenPort'
/usr/local/vitasdk/bin/../lib/gcc/arm-vita-eabi/10.3.0/../../../../arm-vita-eabi/bin/ld: /usr/local/vitasdk/arm-vita-eabi/lib/libSDL2.a(SDL_vitaaudio.c.obj): in function `VITAAUD_CloseDevice':
(.text+0x2e6): undefined reference to `sceAudioInReleasePort'
/usr/local/vitasdk/bin/../lib/gcc/arm-vita-eabi/10.3.0/../../../../arm-vita-eabi/bin/ld: /usr/local/vitasdk/arm-vita-eabi/lib/libSDL2.a(SDL_vitaaudio.c.obj): in function `VITAAUD_CaptureFromDevice':
(.text+0x390): undefined reference to `sceAudioInInput'
collect2: error: ld returned 1 exit status
make[2]: *** [redrectangle] Error 1
make[1]: *** [CMakeFiles/redrectangle.dir/all] Error 2
make: *** [all] Error 2
```

Adding `SceAudioIn_stub` to the CMakeList as proposed in this PR fixes the issue.

isage commented:
This is wrong, please revert. SceAudioIn_stub is linked in by `SDL2::SDL2` target.
You can check that in https://github.com/vitasdk/packages/releases/download/master/sdl2.tar.xz `lib/cmake/SDL2/SDL2staticTargets.cmake`

bucanero commented:
Yet the sample fails to compile. I updated everything yesterday using the SDK update script; please check on your side and see that the linker can't complete

isage commented:
Can't reproduce with freshly installed sdk.

isage commented:
Can you provide output of
`ls $VITASDK/arm-vita-eabi/lib/cmake/SDL2/` and `cat $VITASDK/arm-vita-eabi/lib/cmake/SDL2/SDL2staticTargets.cmake`?

Rinnegatamante commented:
Can't reproduce as well on latest vitasdk, gonna drop the commit.

bucanero commented:
> Can you provide output of `ls $VITASDK/arm-vita-eabi/lib/cmake/SDL2/` and `cat $VITASDK/arm-vita-eabi/lib/cmake/SDL2/SDL2staticTargets.cmake`?

I don't have those `/cmake/` folders, I didn't build the libraries, I just installed the SDK using the `vitasdk-update` script on macOS.

The version information:

```
version_info.txt
Built at 2022-05-22 13:05:28
newlib            80bf6110e5255315dc8de10d579cabc1c6eabddc
pthread-embedded  3ad4d887b84e32a226d9dffa019e4c7b229da785
samples           a4bd64274263e7edbb284f09940817d1cad464bf
vita-headers      40aabd928ab66aa3c0278c1f27abd5170e9f932a
vita-toolchain    175f196ee5c37042a37cc72ead98c86505b95aaa
```

Just to recap, I installed the binary sdk release for macOS from the vitasdk GitHub repository. Then I updated it to the latest, by running the `vitasdk-update` script, that downloaded and extracted some libraries and files.
Last step, I went to the `/samples` directory, and tried to build the redrectangle SDL2 sample.  When I tried `cmake . && make` I've got the linking error mentioned in my first message.

The only way I could build the example was adding the `stub` to the CMake file.

Could be something about the macOS binary release?

bucanero commented:
> Can't reproduce as well on latest vitasdk, gonna drop the commit.

I have shared my `version_info` just in case, maybe I've got an older build?

isage commented:
> I don't have those /cmake/ folders
then sample would fail on cmake step. please check again

bucanero commented:
> > I don't have those /cmake/ folders
> 
> then sample would fail on cmake step. please check again

I'm not building the SDL2 library, the `vitasdk-update` tool downloads the compiled library and headers. I just have the `SDL2.a` and the headers, and that's what the sample is using.

isage commented:
vitasdk-update uses vdpm, which installs sdl2 package, which includes cmake and pkgconfig configs.
Cmake *needs* cmake config, otherwise find_package would fail

bucanero commented:
Here's the cmake info:

```
% ls /usr/local/vitasdk/arm-vita-eabi/lib/cmake/SDL2
SDL2Config.cmake                 SDL2_ttfConfig.cmake             SDL2mainTargets-noconfig.cmake
SDL2ConfigVersion.cmake          SDL2_ttfConfigVersion.cmake      SDL2mainTargets.cmake
SDL2Targets-noconfig.cmake       SDL2_ttfTargets-noconfig.cmake   SDL2staticTargets-noconfig.cmake
SDL2Targets.cmake                SDL2_ttfTargets.cmake            SDL2staticTargets.cmake
```

```
% cat /usr/local/vitasdk/arm-vita-eabi/lib/cmake/SDL2/SDL2staticTargets.cmake
# Generated by CMake

if("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" LESS 2.6)
   message(FATAL_ERROR "CMake >= 2.6.0 required")
endif()
cmake_policy(PUSH)
cmake_policy(VERSION 2.6...3.21)
#----------------------------------------------------------------
# Generated CMake target import file.
#----------------------------------------------------------------

# Commands may need to know the format version.
set(CMAKE_IMPORT_FILE_VERSION 1)

# Protect against multiple inclusion, which would fail when already imported targets are added once more.
set(_targetsDefined)
set(_targetsNotDefined)
set(_expectedTargets)
foreach(_expectedTarget SDL2::SDL2-static)
  list(APPEND _expectedTargets ${_expectedTarget})
  if(NOT TARGET ${_expectedTarget})
    list(APPEND _targetsNotDefined ${_expectedTarget})
  endif()
  if(TARGET ${_expectedTarget})
    list(APPEND _targetsDefined ${_expectedTarget})
  endif()
endforeach()
if("${_targetsDefined}" STREQUAL "${_expectedTargets}")
  unset(_targetsDefined)
  unset(_targetsNotDefined)
  unset(_expectedTargets)
  set(CMAKE_IMPORT_FILE_VERSION)
  cmake_policy(POP)
  return()
endif()
if(NOT "${_targetsDefined}" STREQUAL "")
  message(FATAL_ERROR "Some (but not all) targets in this export set were already defined.\nTargets Defined: ${_targetsDefined}\nTargets not yet defined: ${_targetsNotDefined}\n")
endif()
unset(_targetsDefined)
unset(_targetsNotDefined)
unset(_expectedTargets)


# Compute the installation prefix relative to this file.
get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
if(_IMPORT_PREFIX STREQUAL "/")
  set(_IMPORT_PREFIX "")
endif()

# Create imported target SDL2::SDL2-static
add_library(SDL2::SDL2-static STATIC IMPORTED)

set_target_properties(SDL2::SDL2-static PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;${_IMPORT_PREFIX}/include/SDL2"
  INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:SceGxm_stub>;\$<LINK_ONLY:SceDisplay_stub>;\$<LINK_ONLY:SceCtrl_stub>;\$<LINK_ONLY:SceAppMgr_stub>;\$<LINK_ONLY:SceAudio_stub>;\$<LINK_ONLY:SceAudioIn_stub>;\$<LINK_ONLY:SceSysmodule_stub>;\$<LINK_ONLY:SceIofilemgr_stub>;\$<LINK_ONLY:SceCommonDialog_stub>;\$<LINK_ONLY:SceTouch_stub>;\$<LINK_ONLY:SceHid_stub>;\$<LINK_ONLY:SceMotion_stub>;\$<LINK_ONLY:ScePower_stub>;\$<LINK_ONLY:SceProcessmgr_stub>;\$<LINK_ONLY:m>;-Wl,--no-undefined;\$<LINK_ONLY:>"
)

if(CMAKE_VERSION VERSION_LESS 2.8.12)
  message(FATAL_ERROR "This file relies on consumers using CMake 2.8.12 or greater.")
endif()

# Load information for each installed configuration.
get_filename_component(_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
file(GLOB CONFIG_FILES "${_DIR}/SDL2staticTargets-*.cmake")
foreach(f ${CONFIG_FILES})
  include(${f})
endforeach()

# Cleanup temporary variables.
set(_IMPORT_PREFIX)

# Loop over all imported files and verify that they actually exist
foreach(target ${_IMPORT_CHECK_TARGETS} )
  foreach(file ${_IMPORT_CHECK_FILES_FOR_${target}} )
    if(NOT EXISTS "${file}" )
      message(FATAL_ERROR "The imported target \"${target}\" references the file
   \"${file}\"
but this file does not exist.  Possible reasons include:
* The file was deleted, renamed, or moved to another location.
* An install or uninstall procedure did not complete successfully.
* The installation package was faulty and contained
   \"${CMAKE_CURRENT_LIST_FILE}\"
but not all the files it references.
")
    endif()
  endforeach()
  unset(_IMPORT_CHECK_FILES_FOR_${target})
endforeach()
unset(_IMPORT_CHECK_TARGETS)

# This file does not depend on other imported targets which have
# been exported from the same project but in a separate export set.

# Commands beyond this point should not need to know the version.
set(CMAKE_IMPORT_FILE_VERSION)
cmake_policy(POP)
```


I can't really tell why building the sample on my end fails unless I add `SceAudioIn_stub` to the CmakeList. 😓

bucanero commented:
Just to follow up and close this issue, it was a problem on my end.

After cleaning up my `samples` folder and building everything from scratch (`cmake .` , `make`), it worked just fine. 👍

bucanero commented:
as suggested on Discord, removing `SDL2Targets.cmake` and `SDL2Targets-noconfig.cmake` from `/usr/local/vitasdk/arm-vita-eabi/lib/cmake/SDL2` fixed the issue. 👍
